troposphere/static/js/components/projects/detail/resources/tableData/instance/Name.react.js

```diff
@@ -15,7 +15,7 @@ export default React.createClass({
       var instance = this.props.instance,
           name = instance.get('name').trim() || "[no instance name]";
 
-      if (!instance.id) {
+      if (!instance && !instance.get('id')) {
```

Comment:
I agree - I think `if (instance && instance.isNew())` is a much better way to handle this.